ChatLayout/Classes/Core/Model/ItemModel.swift

```diff
@@ -13,7 +13,9 @@
 import Foundation
 import UIKit
 
+@MainActor
```

Comment:
Is the `@MainActor` truly needed here? I see `resetSize` is set as `nonisolated` later; probably setting `ItemModel` to be `Sendable` should be enough and would lift arbitrary restrictions on `ItemModel`.

Reply:
It is actually possible. I did `nonisolated` just because I reset size in `DispatchQueue.concurrentPerform` so it happens faster. But SectionModel and so on need to be MainActor as they are reading properties from UICollectionViewLayout that is marked as @MainActor. So I do not really se a big reason not to keep an ItemModel a MainActor. But yes - by calling it Sendable allows to remove a `nonisolated` marking.

Reply:
Also, I either do nto understand something about new concurrence or it still sucks.

For example for the next instance it requires MainActor annotation on method, but not it constructror.
<img width="1388" alt="Screenshot 2025-07-03 at 16 48 26" src="https://github.com/user-attachments/assets/107ae008-d69f-400c-bb61-26f0fbfc77e8" />

But then in the very similar case it requires MainActor annotation in constructor. Why is it so I dont understand.
<img width="1349" alt="Screenshot 2025-07-03 at 16 49 22" src="https://github.com/user-attachments/assets/10f55dbc-e5b9-43c7-b4f3-f4028d3f6ccc" />

Reply:
Hm.  But how does compiler know that UICollectionView doesn’t change those things while I am doing switch over them in another thread. Not sure

Reply:
Could be

Reply:
@Frizlab It seems a bug
https://forums.swift.org/t/question-mainactor-in-constructor-is-required-or-not/80891

Reply:
Good to know but I’m not entirely convinced. I’ll follow the thread. Thanks for posting/notifying me!